### PR TITLE
Removed TENSOR_COLUMN_NAME throughout

### PR DESCRIPTION
**Pull request by Bhav00**
## Why are these changes needed?

Related to the issue https://github.com/ray-project/ray/issues/37047. 

Removed the code with `TENSOR_COLUMN_NAME`, a total of 42 references. 

Removed the whole function where parameters/variables the function was working upon were depended on the `TENSOR_COLUMN_NAME`. Otherwise tried to cut only a part of code so as to maintain functionality of the rest even without `TENSOR_COLUMN_NAME` wherever possible.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Bhav00** [comment]
@bveeramani 
Apparently most of the failed cases have the error
> ValueError: Received data of type: <class 'numpy.ndarray'>, but expected it to be one of typing.Union[ForwardRef('numpy.ndarray'), ForwardRef('pandas.DataFrame'), typing.Dict[str, ForwardRef('numpy.ndarray')]]

The function is `_convert_batch_type_to_pandas()` in the file `ray\python\ray\air\util\data_batch_conversion.py`

File: python/ray/air/util/data_batch_conversion.py
Comment (bveeramani) [inline python/ray/air/util/data_batch_conversion.py:68]: Could we keep this code path here? I don't think we can deprecate this quite yet

File: python/ray/air/tests/test_data_batch_conversion.py
Comment (bveeramani) [inline python/ray/air/tests/test_data_batch_conversion.py:204]: I think we need to keep these tests

**Review by bveeramani — CHANGES_REQUESTED** [review]
Thanks for fixing this! I checked with the team, and I think we need to keep some of the code in `python/ray/air`. Otherwise, looks good to me

File: python/ray/train/huggingface/transformers/transformers_predictor.py
Comment (Bhav00) [reply python/ray/train/huggingface/transformers/transformers_predictor.py:290]: > data = data[feature_columns] if feature_columns else data

This code was right below the if statement you mentioned and was doing the same thing as that `if` code, that's why I removed it

File: rllib_contrib/r2d2/README.md
Comment (Bhav00) [reply rllib_contrib/r2d2/README.md:1]: This was pushed to the main repo while I was working on the issue and it updated while I pulled the changes. Changes to that file weren't made by me explicitly.

**Comment by Bhav00** [comment]
@bveeramani I will revert back the three removed tests in `test_data_batch_conversion.py` and the `if` statement in `data_batch_conversion.py`.
These changes would require `TENSOR_COLUMN_NAME` to be put back to the constants file, like it was before.

**Comment by bveeramani** [comment]
> @bveeramani I will revert back the three removed tests in `test_data_batch_conversion.py` and the `if` statement in `data_batch_conversion.py`. These changes would require `TENSOR_COLUMN_NAME` to be put back to the constants file, like it was before.

Thanks! Looks like we need to remove one of the fixtures from the `BatchMapper` test:

```
@pytest.mark.parametrize(
--
  | "ds",
  | [
  | lazy_fixture("ds_pandas_single_column_format"),
  | lazy_fixture("ds_pandas_multi_column_format"),
  | lazy_fixture("ds_pandas_list_multi_column_format"),
  | lazy_fixture("ds_arrow_single_column_format"),
  | lazy_fixture("ds_arrow_single_column_tensor_format"),
  | lazy_fixture("ds_arrow_multi_column_format"),
  | lazy_fixture("ds_list_arrow_multi_column_format"),
  | lazy_fixture("ds_numpy_single_column_tensor_format"),
  | lazy_fixture("ds_numpy_list_of_ndarray_tensor_format"),
  | ],
  | )
  | def test_batch_mapper_batch_size(ray_start_regular_shared, ds):
  | E       fixture 'ds_arrow_single_column_tensor_format' not found
```

**Comment by Bhav00** [comment]
> > @bveeramani I will revert back the three removed tests in `test_data_batch_conversion.py` and the `if` statement in `data_batch_conversion.py`. These changes would require `TENSOR_COLUMN_NAME` to be put back to the constants file, like it was before.
> 
> Thanks! Looks like we need to remove one of the fixtures from the `BatchMapper` test:
> 
> ```
> @pytest.mark.parametrize(
> --
>   | "ds",
>   | [
>   | lazy_fixture("ds_pandas_single_column_format"),
>   | lazy_fixture("ds_pandas_multi_column_format"),
>   | lazy_fixture("ds_pandas_list_multi_column_format"),
>   | lazy_fixture("ds_arrow_single_column_format"),
>   | lazy_fixture("ds_arrow_single_column_tensor_format"),
>   | lazy_fixture("ds_arrow_multi_column_format"),
>   | lazy_fixture("ds_list_arrow_multi_column_format"),
>   | lazy_fixture("ds_numpy_single_column_tensor_format"),
>   | lazy_fixture("ds_numpy_list_of_ndarray_tensor_format"),
>   | ],
>   | )
>   | def test_batch_mapper_batch_size(ray_start_regular_shared, ds):
>   | E       fixture 'ds_arrow_single_column_tensor_format' not found
> ```


This has been done. 
I have one question, I did a manual commit right now to make sure its signed off this time instead of letting vs code handle it, does this kind of mess of 30+ commits happen every now and then whenever someone tries to sign off the commit before pushing or is it only me?

**Comment by Bhav00** [comment]
@bveeramani any updates regarding this pull request please?

**Comment by bveeramani** [comment]
Hey @Bhav00, sorry for the delayed review. I've been busy with the Ray 2.6 release. My teammate @scottjlee will take a look

**Comment by Bhav00** [comment]
> Hey @Bhav00, sorry for the delayed review. I've been busy with the Ray 2.6 release. My teammate @scottjlee will take a look

Alright, please take your time.

**Review by scottjlee — APPROVED** [review]
LGTM overall! Nice work @Bhav00 and thanks for your first contribution! 

Before we merge this, could you do a couple of things:
1) Rebase your branch to master, so that we only see changes relevant to this PR (e.g. removes the unrelated RLLib changes). This will also let us confirm that all relevant tests are passing (currently, there are a few test failures that look unrelated, but we should confirm).
2) Run the linter, so that the lint test passes in the CI. You can find more details [here](https://docs.ray.io/en/latest/ray-contribute/getting-involved.html#lint-and-formatting).

> I have one question, I did a manual commit right now to make sure its signed off this time instead of letting vs code handle it, does this kind of mess of 30+ commits happen every now and then whenever someone tries to sign off the commit before pushing or is it only me?
Oh yeah, this can happen sometimes when the pushed commit does not have a signoff on it. Usually, if I catch this before I push out the commit, I amend the commit with `git commit --amend --signoff` which inserts the signoff into the most recent commit. 

Let us know if you have any questions regarding the above. Once we address the above points, I'll assign this PR to a committer for final merging. Thanks again!

**Comment by Bhav00** [comment]
>Auto-merging doc/source/ray-air/examples/torch_detection.ipynb
>CONFLICT (content): Merge conflict in doc/source/ray-air/examples/torch_detection.ipynb
>error: could not apply 2eae247ef9... [Data][Train] Fix remaining issues on DatasetConfig->DataConfig migration (#37215)

Getting this while trying to rebase the branch, I assume this is one of the many commits during my development timeline. Am I supposed to accept all such changes?

**Comment by Bhav00** [comment]
@scottjlee sorry I forgot to ping

> > Auto-merging doc/source/ray-air/examples/torch_detection.ipynb
> > CONFLICT (content): Merge conflict in doc/source/ray-air/examples/torch_detection.ipynb
> > error: could not apply [2eae247](https://github.com/ray-project/ray/commit/2eae247ef919495931dd73755ebbf52eb608fb34)... [Data][Train] Fix remaining issues on DatasetConfig->DataConfig migration (#37215)
> 
> Getting this while trying to rebase the branch, I assume this is one of the many commits during my development timeline. Am I supposed to accept all such changes?

**Comment by scottjlee** [comment]
Yeah you should merge in upstream changes from master which are appropriate (usually just accept upstream if those changes are not related to yours), and if there are merge conflicts with your code in the PR, keep the appropriate code from both upstream and your PR.

**Comment by Bhav00** [comment]
> Yeah you should merge in upstream changes from master which are appropriate (usually just accept upstream if those changes are not related to yours), and if there are merge conflicts with your code in the PR, keep the appropriate code from both upstream and your PR.

There were only two conflicts and I accepted the incoming changes for both, rest of the rebase process was smooth.
@scottjlee

**Comment by scottjlee** [comment]
@Bhav00 it looks like there are several files which were deleted, which should be kept according to master. For example, `rllib_contrib/r2d2/tests/test_r2d2.py`. Since we are only removing less than 50 occurrences of the constant, I'd expect the overall diff to be in the several hundred lines. Could you verify that the merge keeps the changes from master?

**Comment by Bhav00** [comment]
> @Bhav00 it looks like there are several files which were deleted, which should be kept according to master. For example, `rllib_contrib/r2d2/tests/test_r2d2.py`. Since we are only removing less than 50 occurrences of the constant, I'd expect the overall diff to be in the several hundred lines. Could you verify that the merge keeps the changes from master?

So I took the example of file path that you mentioned to see what's wrong with the branch, the changes done to the main ray repo are being reflected to my fork's master branch, but when it comes to the develop branch named `TENSOR_COLUMN_NAME-removed`, they aren't showing up there, like the `rllib_contrib/r2d2/tests/test_r2d2.py` folder is just not present in local branch but its present in the main branch. I have tried merging these two branches and it does return that they both are up to date but the changes from master aren't being adopted by develop branch.
@scottjlee

**Comment by scottjlee** [comment]
Sorry for the delay on this @Bhav00 , what are the exact commands you are using to merge changes from master branch into your development branch? 

Alternatively, if it's becoming too complicated, it might be quicker to start on a fresh branch, and carry over the changes

**Comment by Bhav00** [comment]
> Sorry for the delay on this @Bhav00 , what are the exact commands you are using to merge changes from master branch into your development branch?
> 
> Alternatively, if it's becoming too complicated, it might be quicker to start on a fresh branch, and carry over the changes that you want to keep.
> 
> This [docs section](https://docs.ray.io/en/latest/ray-contribute/getting-involved.html#submitting-and-merging-a-contribution) provides high level recommendations for working with git.

Uh because the merge said that the branches are up to date, I tried using rebase instead and the documentation you provided definitely says that rebasing is avoided as the commits are squashed during the process.

> Alternatively, if it's becoming too complicated, it might be quicker to start on a fresh branch, and carry over the changes that you want to keep. 
Very much in the favor of this right now, looking at all the mess I have made of a few changes.

**Comment by scottjlee** [comment]
Thanks, please feel free to assign me the new PR once it's ready for review, or if there's anything else I can help with!

**Pull request closed without merging**
